lib/experimental/class-gutenberg-rest-comment-controller.php

```diff
@@ -9,12 +9,123 @@
+		if ( ! empty( $request['context'] ) && 'edit' === $request['context'] && ! current_user_can( $edit_cap ) ) {
+			return new WP_Error(
+				'rest_forbidden_context',
+				__( 'Sorry, you are not allowed to edit comments.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			$protected_params = array( 'author', 'author_exclude', 'author_email', 'type', 'status' );
+			$forbidden_params = array();
+
+			foreach ( $protected_params as $param ) {
+				if ( 'status' === $param ) {
+					if ( 'approve' !== $request[ $param ] ) {
+						$forbidden_params[] = $param;
+					}
+				} elseif ( 'type' === $param ) {
+					if ( 'comment' !== $request[ $param ] ) {
+						$forbidden_params[] = $param;
+					}
+				} elseif ( ! empty( $request[ $param ] ) ) {
+					$forbidden_params[] = $param;
+				}
+			}
+
+			if ( ! empty( $forbidden_params ) ) {
+				return new WP_Error(
+					'rest_forbidden_param',
+					/* translators: %s: List of forbidden parameters. */
+					sprintf( __( 'Query parameter not permitted: %s', 'gutenberg' ), implode( ', ', $forbidden_params ) ),
+					array( 'status' => rest_authorization_required_code() )
+				);
+			}
+		}
+
+		return true;
+	}
+
+	public function get_item_permissions_check( $request ) {
+		$comment = $this->get_comment( $request['id'] );
+		if ( is_wp_error( $comment ) ) {
+			return $comment;
+		}
+
+		// Re-map edit context capabilities when requesting `block_comment` type.
+		// Note: This is only relevant change for the backport.
+		$edit_cap = 'block_comment' === $comment->comment_type ? 'edit_posts' : 'moderate_comments';
+		if ( ! empty( $request['context'] ) && 'edit' === $request['context'] && ! current_user_can( $edit_cap ) ) {
+			return new WP_Error(
+				'rest_forbidden_context',
+				__( 'Sorry, you are not allowed to edit comments.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		$post = get_post( $comment->comment_post_ID );
+
+		if ( ! $this->check_read_permission( $comment, $request ) ) {
+			return new WP_Error(
+				'rest_cannot_read',
+				__( 'Sorry, you are not allowed to read this comment.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( $post && ! $this->check_read_post_permission( $post, $request ) ) {
+			return new WP_Error(
+				'rest_cannot_read_post',
+				__( 'Sorry, you are not allowed to read the post for this comment.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
 		}
 
-		if ( ! is_user_logged_in() ) {
+		return true;
+	}
+
+	public function create_item_permissions_check( $request ) {
+		$is_block_comment = ! empty( $request['comment_type'] ) && 'block_comment' === $request['comment_type'];
+
+		// Note: This is only relevant change for the backport.
+		if ( ! is_user_logged_in() && ! $is_block_comment ) {
```

Comment:
The `block_comment` should always require auth.

Reply:
Nice! Thanks for the inline comment which will help with backporting.  Another comment at the top saying "This code is copied exactly from (core file name) except for sectioned marked with the comment '// Note: This is only relevant change for the backport.'" would be nice.